lib/scope_file_access.rb

```diff
@@ -0,0 +1,58 @@
+module ScopeFileAccessFlat
+  def retrieve_files_from_system(system, paths)
+    system.retrieve_files(paths, scope_file_store.path)
+  end
+
+  def write_file(system_file, target)
+    raise Machinery::Errors::FileUtilsError, "Not a file" if !system_file.file?
+
+    target_path = File.join(target, system_file.name)
+    FileUtils.mkdir_p(File.dirname(target_path))
+    FileUtils.cp(file_path(system_file), target_path)
+  end
+
+  def file_path(system_file)
+    raise Machinery::Errors::FileUtilsError, "Not a file" if !system_file.file?
+
+    File.join(scope_file_store.path, system_file.name)
+  end
+end
+
+module ScopeFileAccessArchive
+  def retrieve_files_from_system_as_archive(system, files, excluded_files)
+    extractor = FileExtractor.new(system, scope_file_store)
+    extractor.extract_files(files, excluded_files)
+  end
+
+  def retrieve_trees_from_system_as_archive(system, trees, excluded_files)
+    extractor = FileExtractor.new(system, scope_file_store)
+    extractor.extract_trees(trees, excluded_files)
+  end
+
+  def export_files_as_tarballs(destination)
```

Comment:
Assignment Branch Condition size for export_files_as_tarballs is too high. [17.15/15]